Better docs for jnp.gradient

jakevdp opened the pull request:
Also remove `skip_params` option from `util.implements`, as this was its last usage.

Rendered preview: https://jax--24025.org.readthedocs.build/en/24025/_autosummary/jax.numpy.gradient.html

Part of #21461

dfm commented on jax/_src/numpy/lax_numpy.py:
```diff
@@ -1451,14 +1451,67 @@ def ediff1d(ary: ArrayLike, to_end: ArrayLike | None = None,
+      - not specified: unit spacing in all dimensions.
+      - a single scalar: constant spacing in all dimensions.
+      - *N* values: specify different spacing in each dimension:
+        - scalar values indicate constant spacing in that dimension.
```
The docs build seems to fail because of this indentation level here.

jakevdp replied:
Fixed, thanks!

dfm commented on jax/_src/numpy/lax_numpy.py:
```diff
@@ -1451,14 +1451,67 @@ def ediff1d(ary: ArrayLike, to_end: ArrayLike | None = None,
+
+  The gradient in ``jnp.gradient`` is computed using second-order finite
+  differences across the array of sampled function values. This should not
+  be confused with :func:`jax.grad`, which computes the autodiff-based
+  gradient of a callable function.
```
Maybe add a link to the intro autodiff tutorial?

jakevdp replied:
Done

jakevdp committed:
Better docs for jnp.gradient

Also remove skip_params option from util.implements, as this was its last usage.
